src/shogun/machine/IterativeLinearMachine.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+* This software is distributed under BSD 3-clause license (see LICENSE file).
+*
+* Authors: Shubham Shukla
+*/
+
+#include <shogun/base/progress.h>
+#include <shogun/lib/common.h>
+#include <shogun/lib/config.h>
+#include <shogun/machine/IterativeLinearMachine.h>
+
+using namespace shogun;
+
+CIterativeLinearMachine::CIterativeLinearMachine() : CLinearMachine()
+{
+	m_current_iteration = 0;
+	m_complete = false;
+
+	SG_ADD(
+	    &m_current_iteration, "current_iteration",
+	    "Current Iteration of training", MS_NOT_AVAILABLE);
+	SG_ADD(
+	    &m_max_iterations, "max_iterations", "Maximum number of Iterations",
+	    MS_AVAILABLE);
+	SG_ADD(&m_complete, "complete", "Convergence status", MS_NOT_AVAILABLE);
+}
+
+CIterativeLinearMachine::~CIterativeLinearMachine()
+{
+	SG_UNREF(m_continue_features);
```

Comment:
what we could do inside train_machine is: 
sgref(m_continue_features)
continue->train()
if(naturally stopped)
sgunref(m_continue_features)
this will cover our cases and we do not need to unref in destructor then.... will that work ?

Reply:
sgref data is not needed if you dont store the pointer (subclasses or base classes might and will take care of this)

calling train machine should first forget old continue features, so the unref seems valid

You will have to try all cases in unit tests to really figure it out, it is sometimes hard to foresee

Reply:
but the rule is simple: 
* always ref if you store a pointer
* always unref in dtor and when you overwrite it
* if overwriting a pointer, first ref the new one and only then unref the old one (in case the obj is the same you avoid a deletion of the obj this way)